### DISCUSSION TITLE
提供中文文档，以及问题解惑

### DISCUSSION THREAD

**Discussion by pandagood** [alibaba/GraphScope#16] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

1，居然没有中文文档，这是专门为外国人开发的吗 ？能不能提供中文文档 ?

2，有些概念不是很懂，网上资料太少，根据介绍，这是个图计算平台，图查询，图深度学习、图分析平台。那么图存储呢？ 数据从哪里来？

这个跟图数据库有什么不同吗？是否具有图数据库的功能？能否当做图数据库来使用 ？像Neo4j、JanusGraph那样。比如我要开发一个社交应用，能否使用GraphScope 作为数据库解决方案？

3，当前只有python客户端，那其他语言，比如go、rust、java等等怎么使用GraphScope ?

**Comment by wenyuanyu** [comment, marked as answer]
感谢对我们项目的兴趣！

1. 我们会提供中文文档（未来1-2个月内）和答疑，欢迎用提问～ 不过我们更加鼓励大家用英文互动，原因是你碰到的问题和解决方案可能有共性，问题并不是说中文用户特有的。 尽量用英文，会方便更大范围用户和开发者互动，也能让项目更成熟、开放，我觉得这也是真正成熟/社区驱动开源项目的普遍情况。

2、3. 我们目前不提供传统“图数据库”意义上“图存储”，但是支持很多外部数据源，对于存储或者Java等多语言的支持我们打算在2021年支持。现在你提的这些还在计划当中，后续在2021年会有更多的更新和支持。不过如果你有具体的case，您可以继续留言，我们会看看现在现状是否能满足？
谢谢你的建议，我们会放一个FAQ来把常见的这些问题来回答一下。

GraphScope是一个结合了阿里很多技术的全新项目，在开源方式下同步开发。有兴趣可以加钉钉群：31533139来跟我们互动。

Thanks for the interests!

1. We will provide support and documents in Chinese. However, English is preferred as we aim to engage users and developers both in and out of China.

2 3. Currently the GraphScope does not contain a mutable storage module (as seen in Neo4j or JanusGraph). It has Python support only. However, GraphScope is evolving fast, and many of the features are planned for the next a few months. If you have any specific application scenarios, please let us know.

**Comment by pandagood** [comment]
@wenyuanyu 感谢解答。我英文不好，英文好的人是少数，中文文档是必须的。

> 我们目前不提供传统“图数据库”意义上“图存储”，但是支持很多外部数据源.    

如果没有图存储的话，那这个无法当做图数据库来使用，只能作为一个生态工具使用。虽然支持外部数据源，但是一般不会以这样的方式作为应用数据库的解决方案。
如果原生支持图存储的话，那么性能上也会比使用外部的数据源要好很多，而且没有兼容性等其他问题，可以当做图数据库使用。
     
> 对于存储或者Java等多语言的支持我们打算在2021年支持

持续关注，希望尽快实现，底层存储最好基于rust开发，同时具备性能和安全。

目前成熟的图数据库极少，之前试过Neo4j、JanusGraph这些，但都不理想，然后又试了[dgraph](https://github.com/dgraph-io/dgraph)，但这个问题很多，基于go，性能虽然比前面的好很多但也有瓶颈。

而现在正在研究[nebula](https://github.com/vesoft-inc/nebula)，一个国产的图数据库，这应该是目前最好的一个，但仍不够完善和稳定，目前也在频繁改版中，而且它底层存储基于rocksdb而不是自己定制开发的，不是最理想的样子。

所以到现在还没有一个完全可用的图数据库，仍在持续关注和等待。o(╯□╰)o


> 不过如果你有具体的case，您可以继续留言，我们会看看现在现状是否能满足

我是做社交应用的，现在的GraphScope还不能满足，我需要的是一个图数据库，而不是一个图分析工具。

期待GraphScope尽快成长起来，最好能提供个路线图时间表🍻

**Comment by wenyuanyu** [comment]
@pandagood 感谢回复！中文没有问题。
我们还在密集的讨论我们的路线图，在合适的时候我们会放出来分享给大家。
图的持久化支持CRUD的存储也是我们很重要的场景。敬请期待！保持联系！常回来看看

**Comment by wenyuanyu** [comment]
Closing this issue.
Opening up #28 and #29 for Chinese documentation and supporting service-like workload respectively.
